File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.js
// Last updated: 10/03/2026, 13:44:59
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let freq = {};

    for (let ch of s) {
        freq[ch] = (freq[ch] || 0) + 1;
    }

    for (let i = 0; i < s.length; i++) {
        if (freq[s[i]] === 1) {
            return i;
        }
    }

    return -1;
};

File: 387-FirstUniqueCharacterinaString/387-FirstUniqueCharacterinaString.cpp
// Last updated: 15/04/2026, 10:52:27
1class Solution {
2public:
3    int firstUniqChar(string s) {
4        vector<int> freq(26, 0);
5
6        for(char c : s){
7            freq[c - 'a']++;
8        }
9
10        for(int i = 0; i < s.size(); i++){
11            if(freq[s[i] - 'a'] == 1){
12                return i;
13            }
14        }
15
16        return -1; 
17    }
18};